src/tbp/monty/frameworks/environments/embodied_data.py

```diff
@@ -336,6 +349,17 @@ def change_object_by_idx(self, idx):
             "semantic_id": self.semantic_label_to_id[self.object_names[idx]],
             **self.object_params,
         }
+        if self.primary_target["object"] in self.parent_to_child_mapping:
+            self.consistent_child_objects = self.parent_to_child_mapping[
+                self.primary_target["object"]
+            ]
+        elif self.parent_to_child_mapping.keys():
```

Comment:
Checking to see if a dictionary is empty is more idiomatically done this way:
```suggestion
        elif self.parent_to_child_mapping:
```
The comment also should be updated or removed. I'm not sure it's needed after this change.